### PR TITLE
Feat(connector): [ELAVON] Template PR

### PR DESCRIPTION
**Pull request by sumanmaji4**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

Template code added for new connector Elavon
https://developer.elavon.com/products/xml-api/v1/api-reference#tag/Process-Transaction-Online

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

File: crates/hyperswitch_connectors/src/default_implementations_v2.rs [hunk 1]
```diff
@@ -236,6 +237,7 @@ macro_rules! default_imp_for_new_connector_integration_refund {
 }
 
 default_imp_for_new_connector_integration_refund!(
+    connectors::Elavon,
```
Comment (deepanshu-iiitu) [inline crates/hyperswitch_connectors/src/default_implementations_v2.rs:240]: Please maintain alphabetical order

**Review by deepanshu-iiitu — CHANGES_REQUESTED** [review]

**Review by deepanshu-iiitu — DISMISSED** [review]

**Review by sai-harsha-vardhan — DISMISSED** [review]

**Review by SanchithHegde — APPROVED** [review]

**Review by cookieg13 — DISMISSED** [review]

**Review by sai-harsha-vardhan — DISMISSED** [review]

**Review by SanchithHegde — DISMISSED** [review]

**Review by deepanshu-iiitu — APPROVED** [review]

File: crates/common_enums/src/enums.rs [hunk 2]
```diff
@@ -142,6 +140,133 @@ pub enum AttemptStatus {
     DeviceDataCollectionPending,
 }
 
+#[derive(
```
Comment (cookieg13) [inline crates/common_enums/src/enums.rs:143]: please remove these extra changes

File: crates/api_models/src/enums.rs [hunk 3]
```diff
@@ -27,6 +27,302 @@ pub enum RoutingAlgorithm {
     Custom,
 }
 
+/// A connector is an integration to fulfill payments
```
Comment (cookieg13) [inline crates/api_models/src/enums.rs:30]: please remove these extra changes

File: crates/api_models/src/enums.rs [hunk 4]
```diff
@@ -1,10 +1,10 @@
 use std::str::FromStr;
 
 pub use common_enums::*;
+#[cfg(feature = "dummy_connector")]
```
Comment (cookieg13) [inline crates/api_models/src/enums.rs:4]: here also

File: crates/api_models/src/enums.rs (hunk 3 above)
Comment (cookieg13) [reply crates/api_models/src/enums.rs:30]: rebase with main

File: crates/api_models/src/enums.rs (hunk 4 above)
Comment (sumanmaji4) [reply crates/api_models/src/enums.rs:4]: @cookieg13 I encountered some merge conflicts due to these changes, so it seems they may all be needed going forward.

**Review by deepanshu-iiitu — APPROVED** [review]

**Review by cookieg13 — APPROVED** [review]

**Review by sai-harsha-vardhan — APPROVED** [review]

**Review by SanchithHegde — APPROVED** [review]

**Pull request merged**